### PR TITLE
HOSTEDCP-1525: Support Azure KMS key rotation

### PR DESCRIPTION
**Pull request by muraee**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #[HOSTEDCP-1525](https://issues.redhat.com/browse/HOSTEDCP-1525)

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: api/hypershift/v1beta1/hostedcluster_types.go [hunk 1]
```diff
@@ -1938,9 +1938,15 @@ type KMSSpec struct {
 
 // AzureKMSSpec defines metadata about the configuration of the Azure KMS Secret Encryption provider using Azure key vault
 type AzureKMSSpec struct {
```
Comment (bryan-cox) [inline api/hypershift/v1beta1/hostedcluster_types.go:1940]: Do we need additional CEL here? Is this required?

Comment (muraee) [reply api/hypershift/v1beta1/hostedcluster_types.go:1940]: yes, add required validation.

**Pull request merged**